src/zh/komiic/src/eu/kanade/tachiyomi/extension/zh/komiic/Komiic.kt

```diff
@@ -0,0 +1,414 @@
+    }
+
+    private fun parseMangaIDList(response: Response): MangasPage {
+        val mangas = mutableListOf<SManga>()
+        val jsonData = response.body.string()
+        val jsonObj = JSONObject(jsonData)
+        val jsonDict = jsonObj.getJSONObject("data").getJSONObject("comicById")
+
+        val manga = SManga.create().apply {
+            title = jsonDict.getString("title")
+            thumbnail_url = jsonDict.getString("imageUrl")
+            url = "/comic/${jsonDict.getString("id")}"
+            author = parseAuthors(jsonDict.getJSONArray("authors"))
+            genre = parseCategories(jsonDict.getJSONArray("categories"))
+            status = parseStatus(jsonDict.getString("status"))
+        }
+        mangas.add(manga)
+
+        val hasNextPage = jsonDict.length() >= 20
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        val data = buildJsonObject {
+            put("operationName", "hotComics")
+            putJsonObject("variables") {
+                putJsonObject("pagination") { put("limit", 20); put("offset", (page - 1) * 20); put("orderBy", "VIEWS"); put("status", ""); put("asc", true) }
+            }
+            put("query", "query hotComics(\$pagination: Pagination!) {\n  hotComics(pagination: \$pagination) {\n    id\n    title\n    status\n    year\n    imageUrl\n    authors {\n      id\n      name\n      __typename\n    }\n    categories {\n      id\n      name\n      __typename\n    }\n    dateUpdated\n    monthViews\n    views\n    favoriteCount\n    lastBookUpdate\n    lastChapterUpdate\n    __typename\n  }\n}")
+        }
+
+        return client.newCall(POSTJson(queryAPI, data))
+            .asObservableSuccess()
+            .map { response ->
+                parseMangaList(response, "hotComics")
+            }
+    }
+
+    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> {
+        val data = buildJsonObject {
+            put("operationName", "recentUpdate")
+            putJsonObject("variables") {
+                putJsonObject("pagination") { put("limit", 20); put("offset", (page - 1) * 20); put("orderBy", "DATE_UPDATED"); put("status", ""); put("asc", true) }
+            }
+            put("query", "query recentUpdate(\$pagination: Pagination!) {\n  recentUpdate(pagination: \$pagination) {\n    id\n    title\n    status\n    year\n    imageUrl\n    authors {\n      id\n      name\n      __typename\n    }\n    categories {\n      id\n      name\n      __typename\n    }\n    dateUpdated\n    monthViews\n    views\n    favoriteCount\n    lastBookUpdate\n    lastChapterUpdate\n    __typename\n  }\n}")
+        }
+
+        return client.newCall(POSTJson(queryAPI, data))
+            .asObservableSuccess()
+            .map { response ->
+                parseMangaList(response, "recentUpdate")
+            }
+    }
+
+    override fun fetchSearchManga(
```

Comment:
generally avoid overriding `fetch` methods, and simply use the `request` and corresponding `parse` methods, unless required